runtime/src/snapshot_utils/snapshot_storage_rebuilder.rs

```diff
@@ -317,10 +317,29 @@ impl SnapshotStorageRebuilder {
             })
             .collect::<Result<HashMap<AppendVecId, Arc<AccountStorageEntry>>, std::io::Error>>()?;
 
-        assert_eq!(slot_stores.len(), 1);
-        let (id, storage) = slot_stores.drain().next().unwrap();
-        self.storage
-            .insert(slot, AccountStorageReference { id, storage });
+        let storage = if slot_stores.len() > 1 {
+            let remapped_append_vec_id = self.next_append_vec_id.fetch_add(1, Ordering::AcqRel);
```

Comment:
nit: maybe it worth adding a log for the ids of old appendvec that are combined to the new one?